RestaurantsNearMe/Data/CloudKitService.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  CloudKitService.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-11-27.
+//
+
+import Foundation
+import Combine
+import CloudKit
+
+class CloudKitService: CloudKitServiceProviding {
+  let applicationContainer = CKContainer.default()
+  private let _fetchingStatus = PassthroughSubject<Bool, Never>()
+
+  private let _accountStatus = CurrentValueSubject<CKAccountStatus, Never>(.couldNotDetermine)
+  var accountStatus: AnyPublisher<CKAccountStatus, Never> {
+    _accountStatus.eraseToAnyPublisher()
+  }
+  
+  init() {
+    Task.init {
+      try? await refreshAccountStatus()
+    }
+  }
+  
+  func fetchAPIKeyByID() async throws {
+    let id = CKRecord.ID(recordName: APIKey.name)
+    guard _accountStatus.value == .available else {
+      throw CloudKitServiceError.containerNotAvailable
+    }
+    
+    do {
+      let apiRecord = try await applicationContainer.publicCloudDatabase.record(for: id)
+      guard let recordValue = apiRecord.object(forKey: APIKeyNames.keyName.rawValue) as? String else {
+        throw CloudKitServiceError.incorrectDataFormat
+      }
+
+      APIKeyProvider.key = APIKey(keyName: recordValue)
+    } catch {
+      throw CloudKitServiceError.fetchFailed
+    }
+  }
+  
+  func refreshAccountStatus() async throws {
+    defer { _fetchingStatus.send(false) }
```

Comment:
This is an OK start, but you may want to introduce a lock or sync thread to control updates - it also may not be guaranteed to be called on the main thread as you haven't added checks or `@MainActor`.